devicetypes/heltun/heltun-hls01-switch.src/heltun-hls01-switch.groovy

```diff
@@ -0,0 +1,302 @@
+ *
+ */
+metadata {
+	definition (name: "HELTUN HLS01 Switch", namespace: "HELTUN", author: "Sarkis Kabrailian", cstHandler: true) {
+		capability "Energy Meter"
+		capability "Power Meter"
+		capability "Switch"
+		capability "Temperature Measurement"
+		capability "Voltage Measurement"
+		capability "Configuration"      
+		capability "Health Check" 
+		capability "Refresh"
+
+		fingerprint mfr: "0344", prod: "0004", inClusters:"0x5E,0x85,0x59,0x8E,0x55,0x86,0x72,0x5A,0x73,0x81,0x87,0x98,0x9F,0x6C,0x70,0x25,0x31,0x32,0x71,0x22,0x7A", deviceJoinName: "HELTUN Switch" //model: "000A"
+	}															
+	preferences {
+		input (
+			title: "HE-HLS01 | HELTUN High Load Switch",
+			description: "The user manual document with all technical information is available in support.heltun.com page. In case of technical questions please contact HELTUN Support Team at support@heltun.com",
+			type: "paragraph",
+			element: "paragraph"
+		)
+		parameterMap().each { 
+			input (
+				title: "${it.title}",
+				description: it.description,
+				type: "paragraph",
+				element: "paragraph"
+			)
+			def unit = it.unit ? it.unit : ""
+			def defV = it.default as Integer
+			def defVDescr = it.options ? it.options.get(defV) : "${defV}${unit} - Default Value" 
+			input (
+				name: it.name,
+				title: null,
+				description: "$defVDescr",
+				type: it.type,
+				options: it.options,
+				range: (it.min != null && it.max != null) ? "${it.min}..${it.max}" : null,
+				defaultValue: it.default,
+				required: false
+			)
+		}
+	}
+}
+
+def initialize() {
+	runIn(3, "checkParam")
+}
+
+def parse(String description) {
+	def result = null
+	def cmd = zwave.parse(description)
+	if (cmd) {result = zwaveEvent(cmd)}
+	return result
+}
+
+def updated() {
+	initialize()
+}
+
+def checkParam() {
+	boolean needConfig = false
+	parameterMap().each {
+		if (state."$it.name" == null || state."$it.name".state == "defNotConfigured") {
+			state."$it.name" = [value: it.default as Integer, state: "defNotConfigured"]
+			needConfig = true
+		}
+		if (settings."$it.name" != null && (state."$it.name".value != settings."$it.name" as Integer || state."$it.name".state == "notConfigured")) {
+			state."$it.name".value = settings."$it.name" as Integer
+			state."$it.name".state = "notConfigured"
+			needConfig = true
+		}
+	}
+	if ( needConfig ) {
+		configParam()
+	}
+}
+
+private configParam() {
+	def cmds = []
+	for (parameter in parameterMap()) {
+	if ( state."$parameter.name"?.value != null && state."$parameter.name"?.state in ["notConfigured", "defNotConfigured"] ) { 
+			cmds << new physicalgraph.device.HubAction(zwave.configurationV2.configurationSet(scaledConfigurationValue: state."$parameter.name".value, parameterNumber: parameter.paramNum, size: parameter.size).format())
+			cmds << new physicalgraph.device.HubAction(zwave.configurationV2.configurationGet(parameterNumber: parameter.paramNum).format())
+			break
+		}
+	}
+	if (cmds) {
+		runIn(5, "checkParam")
```

Comment:
Are You sure that calling this function will not cause any unnecessary configuration loops during device inclusion ?

Reply:
Yes I'm sure